test/session_pool/session_pool.test.js

```diff
@@ -116,24 +116,24 @@ describe('SessionPool - testing session pool', () => {
             expect(isCalled).toBe(true); //eslint-disable-line
         });
 
-        test('should delete picked session when it is usable a create a new one', async () => {
+        test('should delete picked session when it is unusable and create a new one', async () => {
             sessionPool.maxPoolSize = 1;
             await sessionPool.getSession();
             const session = sessionPool.sessions[0];
 
             session.errorScore += session.maxErrorScore;
             let isCalled = false;
-            const oldRemove = sessionPool._removeSession; //eslint-disable-line
+            const oldRemove = sessionPool._removeRetiredSessions; //eslint-disable-line
 
-            sessionPool._removeSession = (session) => { //eslint-disable-line
+            sessionPool._removeRetiredSessions = (session) => { //eslint-disable-line
```

Comment:
I know that "this was already there", but it's a good practice to fix things that you need to change anyway. Instead of using this function replacement, I'd use jest mocking.

Reply:
I don't really even need mocking here. I could test it purely with `.getSession()` - updated